contracts/contracts/strategies/balancer/BalancerMetaPoolStrategy.sol

```diff
@@ -8,14 +8,32 @@ pragma solidity ^0.8.0;
 import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
 import { BaseAuraStrategy, BaseBalancerStrategy } from "./BaseAuraStrategy.sol";
 import { IBalancerVault } from "../../interfaces/balancer/IBalancerVault.sol";
-import { IRateProvider } from "../../interfaces/balancer/IRateProvider.sol";
-import { IMetaStablePool } from "../../interfaces/balancer/IMetaStablePool.sol";
 import { IERC20, InitializableAbstractStrategy } from "../../utils/InitializableAbstractStrategy.sol";
 import { StableMath } from "../../utils/StableMath.sol";
 
 contract BalancerMetaPoolStrategy is BaseAuraStrategy {
     using SafeERC20 for IERC20;
     using StableMath for uint256;
+    /* For Meta stable pools the enum value should be "2" as it is defined 
+     * in the IBalancerVault. From the Metastable pool codebase:
+     * 
+     * enum ExitKind { EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, EXACT_BPT_IN_FOR_TOKENS_OUT, BPT_IN_FOR_EXACT_TOKENS_OUT }
+
+     * For Composable stable pools using IBalancerVault.WeightedPoolExitKind is not
+     * ok since the enum values are in different order as they are in MetaStable pools.
+     * From the pool code: 
+     * 
+     * enum ExitKind { EXACT_BPT_IN_FOR_ONE_TOKEN_OUT, BPT_IN_FOR_EXACT_TOKENS_OUT, EXACT_BPT_IN_FOR_ALL_TOKENS_OUT }
+     */
+    uint256 internal balancerBptInExactTokensOutIndex;
```

Comment:
Could these be immutable set in a constructor?

Reply:
Thanks, very good point. I should have though of that... It clears up code nicely and since each implementing contract will have this populated at creation time, there is no storage slot state/collisions to check. 

fix commit: https://github.com/OriginProtocol/origin-dollar/commit/c21d6f44b1342f75b14af5e5bb2504e4363113a9